### PR TITLE
Undefined behaviour fixes

### PR DESCRIPTION
**Pull request by necessarily-equal**

### CODE REVIEW

File: src/sgame/sg_combat.cpp [hunk 1]
```diff
@@ -158,19 +158,25 @@ static const gentity_t *G_FindKillAssist( const gentity_t *self, const gentity_t
...
+	if ( killer - g_entities >= MAX_CLIENTS )
 	{
 		return nullptr;
 	}
 
-	// Require that the assist was for, at least, 25% of the damage or
+	// Require that the assist was for, at least, 25% of the health or
 	// as much damage as the killer did, whichever is lower
 	damage = self->entity->Get<HealthComponent>()->MaxHealth() / 4.0f;
 	if ( killer )
```
Comment (slipher) [inline src/sgame/sg_combat.cpp:174]: I don't see why there shouldn't be assists when the killer is a non-client (or the same player who died for that matter...). Why not just change `if ( killer )` to `if ( killer && killer->client )`

Comment (necessarily-equal) [reply src/sgame/sg_combat.cpp]: Fair enough, it makes sense to have kill assist messages when something is killed by, say, a turret. Fixed.

File: src/cgame/cg_rocket_datasource.cpp [hunk 2]
```diff
@@ -765,8 +765,14 @@ static void AddToAlOutputs( char *name )
 
 void CG_Rocket_SetAlOutputsOutput( const char*, int index )
 {
+	const char *device = "";
+	if (index >= -1 && index < rocketInfo.data.alOutputsCount )
+	{
+		device = rocketInfo.data.alOutputs[ index ];
```
Comment (slipher) [inline src/cgame/cg_rocket_datasource.cpp:771]: I think `index >= -1` is not what you wanted

Comment (necessarily-equal) [reply src/cgame/cg_rocket_datasource.cpp:771]: Indeed

**Comment by slipher** [comment]
Thanks!

**Pull request merged**